app/models/post.rb

```diff
@@ -57,6 +57,34 @@ def address
     [street, city, state, postal_code, country].compact.join(", ")
   end
 
+  def self.filter_type(type)
+    if type == "needed" || type == "available"
+      send(type.to_sym)
+    else
+      all
```

Comment:
I don't know why I didn't realize this haha. Probably would have answered that immediately by opening a terminal :P.